valhalla/thor/costmatrix.h

```diff
@@ -131,43 +146,37 @@ class CostMatrix {
   std::shared_ptr<sif::DynamicCost> costing_;
 
   uint32_t max_reserved_labels_count_;
+  bool clear_reserved_memory_;
 
   // Number of source and target locations that can be expanded
-  uint32_t source_count_;
-  uint32_t remaining_sources_;
-  uint32_t target_count_;
-  uint32_t remaining_targets_;
+  std::array<uint32_t, 2> locs_count_;
+  std::array<uint32_t, 2> locs_remaining_;
 
   // The cost threshold being used for the currently executing query
   float current_cost_threshold_;
 
   // Status
-  std::vector<LocationStatus> source_status_;
-  std::vector<LocationStatus> target_status_;
-
-  // Adjacency lists, EdgeLabels, EdgeStatus, and hierarchy limits for each
-  // source location (forward traversal)
-  std::vector<std::vector<sif::HierarchyLimits>> source_hierarchy_limits_;
-  std::vector<baldr::DoubleBucketQueue<sif::BDEdgeLabel>> source_adjacency_;
-  std::vector<std::vector<sif::BDEdgeLabel>> source_edgelabel_;
-  std::vector<EdgeStatus> source_edgestatus_;
-
-  // Adjacency lists, EdgeLabels, EdgeStatus, and hierarchy limits for each
-  // target location (reverse traversal)
-  std::vector<std::vector<sif::HierarchyLimits>> target_hierarchy_limits_;
-  std::vector<baldr::DoubleBucketQueue<sif::BDEdgeLabel>> target_adjacency_;
-  std::vector<std::vector<sif::BDEdgeLabel>> target_edgelabel_;
-  std::vector<EdgeStatus> target_edgestatus_;
+  std::array<std::vector<LocationStatus>, 2> locs_status_;
+
+  // Adjacency lists, EdgeLabels, EdgeStatus, and hierarchy limits for each location
+  std::array<std::vector<std::vector<sif::HierarchyLimits>>, 2> hierarchy_limits_;
+  std::array<std::vector<baldr::DoubleBucketQueue<sif::BDEdgeLabel>>, 2> adjacency_;
+  std::array<std::vector<std::vector<sif::BDEdgeLabel>>, 2> edgelabel_;
+  std::array<std::vector<EdgeStatus>, 2> edgestatus_;
 
   // List of best connections found so far
   std::vector<BestCandidate> best_connection_;
 
+  bool ignore_hierarchy_limits_;
```

Comment:
we really have to be more strict when we implement smth for one algorithm, to not leave out all others. at the very least, open an issue or even better, create a "consistency" project or so.